### ISSUE TITLE
Email adds to Per-Product OR ActiveCampaign settings list, not both

### ISSUE THREAD

**Issue by dgoldak** [easydigitaldownloads/edd-activecampaign#25]
### Expected behavior
My assumption is when a list is selected in the per-product list, email address should be added to that list AND the ActiveCampaign settings selected list.

### Actual behavior
If a per-product list is selected, it's only added to that list(s), not the list selected in the settings.

### Steps to reproduce the behavior
1) Select an ActiveCampaign list in the per-product settings, but do not select the same list you have selected in the ActiveCampaign settings.
2) Opt-in upon checkout to newsletter
3) Verify ActiveCampaign, this list should be added to the per-product list(s) and the ActiveCampaign settings selected list.

### Information (if a specific version is affected):
PHP Version: 7.0

EDD Version (or branch): `master`

WordPress Version: 5.0

**Comment by dgoldak** [comment]
My assumption is based on MailChimp's functionality. Technically this may not be a bug if we want to consider this working as expected. If you want customers added to the "Default" list, you can select that list in the per-product settings.

**Comment by cklosowski** [comment]
No this is a bit different. Basically our ActiveCampaign extension is doing the subscription fired on each download on a purchase being completed, instead of the purchase itself.

Our Mailchimp extension is only doing this on payment completion (action `edd_complete_purchase`) and then using a `foreach` on each purchased download, while ActiveCampaign is firing on on each download when the payment is being completed (action `edd_complete_download_purchase`).

The main issue here is that the AC extension is doing this:
```
				// No Download list set so return global list ID
				$list_id = edd_get_option( 'eddactivecampaign_list', false );
				if( ! $list_id ) {
					if( function_exists( 'edd_debug_log' ) ) {
						edd_debug_log( 'ActiveCampaign Debug - List Check. No global site list ID defined, exiting.' );
					}
					return false;
				}
```

which basically says if we don't find a download specific list, just subscribe them to the default list. So if you're testing included a single product with a download specific list, it would only do the download list. If you did a purchase with multiple products, one of which does not have a specific list, you would have been subscribed to the list of the first product (which does have a list assigned) _and_ the default.

We'll just need to make it so that the default list is always subscribed, and then the download specific ones as well.

**Comment by cklosowski** [comment]
@dgoldak that's a good question for @easydigitaldownloads/core-team 

Do we want our maintained email extensions work the same or do we want to keep them working differently?

**Comment by mintplugins** [comment]
I feel like the ideal would be to have them all work the same. But maybe once you get down to it, certain people have chosen certain mailing system because of the way they work, and if we aren't getting any complaints, maybe people are happy. On that train of thought, I'd probably say that we only change it in a response to a customer complaint if/when it arises.

**Comment by cklosowski** [comment]
@mintplugins I'm kind of thinking the same thing. Since this was just an observation, maybe we leave it as for now? I have some code written for it to make it work the same but like you said, if we haven't had any complaints, then why change the behavior.

**Comment by pippinsplugins** [comment]
I agree that ideally they should all function the same way, but at the same time if there are no issues being brought up by customers, let's not give them a reason to raise one.

**Comment by cklosowski** [comment]
So @dgoldak knowing that's how it is coded to work and we don't want to change anything, is this issue something we can consider a 'feature request'?

**Comment by dgoldak** [comment]
@cklosowski yes, that sounds good to me!
